elastalert/loaders.py

```diff
@@ -240,28 +240,46 @@ def load_yaml(self, filename):
             'rule_file': filename,
         }
 
-        self.import_rules.pop(filename, None)  # clear `filename` dependency
-        files_to_import = []
+        current_path = filename
+
+        # Imports are applied using a Depth First Search (DFS) traversal.
+        # If a rule defines multiple imports, both of which define the same value,
+        # the value from the later import will take precedent.
+        import_paths_stack = []
+
         while True:
-            loaded = self.get_yaml(filename)
+            loaded = self.get_yaml(current_path)
 
             # Special case for merging filters - if both files specify a filter merge (AND) them
             if 'filter' in rule and 'filter' in loaded:
                 rule['filter'] = loaded['filter'] + rule['filter']
 
             loaded.update(rule)
             rule = loaded
-            if 'import' in rule:
-                # add all of the files to load into the load queue
-                files_to_import += self.get_import_rule(rule)
-                del (rule['import'])  # or we could go on forever!
-            if len(files_to_import) > 0:
-                # set the next file to load
-                next_file_to_import = files_to_import.pop()
-                rules = self.import_rules.get(filename, [])
-                rules.append(next_file_to_import)
-                self.import_rules[filename] = rules
-                filename = next_file_to_import
+
+            if 'import' not in rule:
+                # clear import_rules cache for current path
+                self.import_rules.pop(current_path, None)
```

Comment:
In what scenario will this `self.import_rules.pop` statement actually pop something?

Reply:
When the `import` is removed from a file.  Added test case to cover this:

https://github.com/jertel/elastalert2/pull/580/commits/ef062016b7622cb8b5125aa8c867dc6139002c5a#diff-68599694211dc95012d74a98b54fd5f439eddb0d4ce257ba1a35a1772f243069R617

Reply:
I see now. It's great to have all this new test coverage, thanks for adding it.

Reply:
Have you automated your rule updates?

Reply:
Yes.  We commit our rules to a git repository and I run cron job inside the docker container to update them.